### ISSUE TITLE
Update actions to upload a package to a CDN

### ISSUE THREAD

**Issue by makermelissa** [adafruit/Adafruit_WebSerial_ESPTool#46]
This is how it was being done by @NabuCasa and after some trial and error, it would probably be best to restore that, but under our control. By the way, great job with all of the setup work. It really is the right way to go. Here's why I didn't do it initially.

When I updated here, I attempted to host the package on GitHub, my goal was to try and have the package hosted here as to not mess around with having an Adafruit NPM account similar to how we did https://github.com/adafruit/ble-file-transfer-js. Additionally, there was a second step of updating what version the web client was pointed to. However, with TypeScript, the source needs to be compiled. Currently the GitHub actions is compiling and uploading the compiled files to gitHub pages on any new push, which makes updates here very easy. However, it also removes versioning. It also makes linking to the node package externally more difficult.

While we could certainly link to the compiled GitHub files directly, it wouldn't be able to go through a CDN like unpkg.com, which is what the WipperSnapper Firmware uploader is currently using (https://github.com/adafruit/WipperSnapper_Firmware_Uploader/blob/main/js/script.js#L6). As such, I haven't felt comfortable linking it to this repo, which means potential loss of control.

Another thing that this affects is something that @tannewt and I have discussed a bit, which is adding automatic firmware deployment to circuitpython.org for ESP32-based devices. This would make it much easier for the user and having the ESPTool as an NPM package would be perfect for making the process easier.

So the steps to fix this are:
- [x] Figure out if we have or who controls an Adafruit NPM Account
- [x] Get the credentials or at least some kind of token that we can use for GitHub Actions
- [x] Update GitHub Actions to upload to account
- [x] Publish to NPM
- [x] Update the source files in this repo to load the new NPM Package
- [x] Test that everything is working
- [x] Update the WipperSnapper_Firmware_Uploader repo to make use of the new NPM Package (or at least create an issue) plus any ancillary related tasks

**Comment by makermelissa** [comment]
Also, once https://github.com/espressif/esptool-js is more stable (currently on version 0.1-dev), it would make sense to start using that. 

However, users (including myself) have run into numerous problems whereas this project worked fine. As a note, I did a quick test and this time I didn't experience the connection problems I had previously run into. I tried programming the board and got the usual timeout error. It looks like somebody created an issue for it (https://github.com/espressif/esptool-js/issues/47) so maybe once that's fixed, maybe it would be worth considering at that point.

**Comment by makermelissa** [comment]
Ok, I think this repo may be ready for retirement very soon. It appears once https://github.com/espressif/esptool-js/pull/83 is merged in that the esptool-js will be ready for some more extensive testing and possibly ready to recommend over this tool. If further testing appears to go well, I'll retire this rather than update it.

**Comment by makermelissa** [comment]
I have discovered a new way that works great. That is to use the git-auto-commit-action GitHub Action. I implemented it in https://github.com/adafruit/web-firmware-installer-js and it basically will commit a folder to the repo. This means a release will have the files and a CDN can access the compiled files. It has all the advantages of the way it is done here and all the advantages of using NPM without using much in the way of external resources.
